generators/pythonv2/codegen/src/ast/Type.ts

```diff
@@ -0,0 +1,211 @@
+import { assertNever } from "@fern-api/core-utils";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+
+type InternalType = Int | Float | Bool | Str | Bytes | List | Set | Tuple | Dict | None | Optional | Union | Any;
+
+interface Int {
+    type: "int";
+}
+
+interface Float {
+    type: "float";
+}
+
+interface Bool {
+    type: "bool";
+}
+
+interface Str {
+    type: "str";
+}
+
+interface Bytes {
+    type: "bytes";
+}
+
+interface Date {
+    type: "date";
+}
+
+interface DateTime {
+    type: "datetime";
+}
+
+interface List {
+    type: "list";
+    value: Type;
+}
+
+interface Set {
+    type: "set";
+    value: Type;
+}
+
+interface Tuple {
```

Comment:
[q] we may want to include a flag here for ellipsis, eg `Tuple[str, ...]`

Reply:
Gonna OOS this from here.